### ISSUE TITLE
gjs failed to build on Ubuntu 14.04

### ISSUE THREAD

**Issue by rwhogg** [Linuxbrew/linuxbrew#767]
gist-log: https://gist.github.com/76ed51ae4d80ea3e5ee8

**Comment by sjackman** [comment]
https://gist.github.com/rwhogg/76ed51ae4d80ea3e5ee8#file-05-make-L184
```
/tmp/gjs20160131-22213-1hqwh2l/gjs-1.44.0/tmp-introspecthly41b/.libs/lt-WarnLib-1.0: error while loading shared libraries: libmozjs-24.so: cannot open shared object file: No such file or directory
Command '['/tmp/gjs20160131-22213-1hqwh2l/gjs-1.44.0/tmp-introspecthly41b/WarnLib-1.0', '--introspect-dump=/tmp/gjs20160131-22213-1hqwh2l/gjs-1.44.0/tmp-introspecthly41b/functions.txt,/tmp/gjs20160131-22213-1hqwh2l/gjs-1.44.0/tmp-introspecthly41b/dump.xml']' returned non-zero exit status 127
Makefile:2634: recipe for target 'WarnLib-1.0.gir' failed
```

`gjs` installs its own copy of `mozjs`. Adding its library path to `LD_LIBRARY_PATH` may work. Try the following untested patch:
```diff
diff --git a/Library/Formula/gjs.rb b/Library/Formula/gjs.rb
index bd5ad2b..74840ae 100644
--- a/Library/Formula/gjs.rb
+++ b/Library/Formula/gjs.rb
@@ -42,6 +42,8 @@ class Gjs < Formula
         rm Dir["#{bin}/*"]
       end
       ENV.append_path "PKG_CONFIG_PATH", "#{lib}/pkgconfig"
+      # Fix error while loading shared libraries: libmozjs-24.so
+      ENV.prepend_path "LD_LIBRARY_PATH", lib if OS.linux?
     end
     system "./configure", "--disable-debug",
                           "--disable-dependency-tracking",
```

**Comment by rwhogg** [comment]
It works. I'm about to submit a PR.

Thanks a bunch!

**Comment by sjackman** [comment]
Great. Glad to hear it. This issue could also affect other formula. It affects `mcrypt` for example: https://github.com/Linuxbrew/linuxbrew/blob/master/Library/Formula/mcrypt.rb#L32

Could you try the following untested patch?

```diff
diff --git a/Library/Homebrew/extend/ENV/std.rb b/Library/Homebrew/extend/ENV/std.rb
index 6fd01b3..cda65d5 100644
--- a/Library/Homebrew/extend/ENV/std.rb
+++ b/Library/Homebrew/extend/ENV/std.rb
@@ -56,6 +56,9 @@ module Stdenv
 
     append "LDFLAGS", "-Wl,-headerpad_max_install_names" if OS.mac?
 
+    # Add this formula's library directory to the shared library search path.
+    prepend_path "LD_LIBRARY_PATH", formula.lib if formula && OS.linux?
+
     if OS.linux? && !["glibc", "glibc25"].include?(formula && formula.name)
       # Set the dynamic linker
       glibc = Formula["glibc"] rescue nil
```

**Comment by rwhogg** [comment]
In progress; will test on both gjs and mcrypt. Let me know if there's any others you come up with.

**Comment by rwhogg** [comment]
It doesn't seem to fix gjs. But the patch does work for mcrypt.

**Comment by sjackman** [comment]
Well that's weird.

**Comment by sjackman** [comment]
Argh. `prepend_path` is weird. It doesn't add the path unless it exists, and `libdir` doesn't yet exist. `prepend` works as expected.
```ruby
    prepend "LD_LIBRARY_PATH", formula.lib, File::PATH_SEPARATOR if formula && OS.linux?
```

**Comment by sjackman** [comment]
Should be fixed now by #771. Thanks for the bug report, Bob.
